### DISCUSSION TITLE
Right-aligned Stacks

### DISCUSSION THREAD

**Discussion by jaakerisalu** [twilio-labs/paste#2078] [Ideas] [answered]
Hey. In Admiral we've been using Stacks to hold our Buttons, but it really backfired when the stack is supposed to be visually right-aligned when the screen width changes and line breaks happen (This came up during a responsive layout task).

Case study:

Wide screen smooth sailing:
![Screen Shot 2021-11-29 at 13 40 21](https://user-images.githubusercontent.com/9283537/143862004-ef601b8e-45ec-4aca-b596-29f911a35f55.png)

Cringeville, USA:
![Screen Shot 2021-11-29 at 13 40 39](https://user-images.githubusercontent.com/9283537/143862072-8bf3e457-99c6-4ac1-957d-9e908aecb0e9.png)

I split the Stack into 2 sub-Stacks so it breaks evenly at once in this case. 
Looking back it might not be the best idea, but the issue is still the same with plain single Stack behavior:
![Screen Shot 2021-11-29 at 13 45 49](https://user-images.githubusercontent.com/9283537/143862539-db600698-caac-4eb4-9a5f-902925531ba1.png)

My idea for your consideration:

Add some sort of `align='right'` or something to Stacks. All it would do is swap the `justify-content` from `flex-start` to `flex-end`, and reverse which side the gutters are on the stack children.

Yours Truly,
Jaak from PLEX

**Comment by kylegilbreath** [comment]
Hi @jaakerisalu, thanks for the question! I've just reached out to our team internally to chat about this. I'll follow up on this discussion when I have more information on this.

**Reply by kylegilbreath** [reply]
Hi @jaakerisalu, just following up. Thanks for the suggestion. This definitely seems like a reasonable request, and an improvement on the Stack component that others would find useful. We've created a [ticket](https://issues.corp.twilio.com/browse/DSYS-3156) on our backlog to add support for this, and will follow up when we've gotten a chance to add this functionality.

**Comment by loreina** [comment, marked as answer]
Marking this as answered due to inactivity. If you'd like to chat about this further, comment here or create a new discussion and our team will get back to you. Thanks!
